airbyte-cdk/python/connector_builder/main.py

```diff
@@ -19,25 +19,36 @@ def create_source(config: Mapping[str, Any]) -> ManifestDeclarativeSource:
     return ManifestDeclarativeSource(manifest, True)
 
 
-def get_config_and_catalog_from_args(args: List[str]) -> Tuple[Mapping[str, Any], ConfiguredAirbyteCatalog]:
+def get_config_and_catalog_from_args(args: List[str]) -> Tuple[str, Mapping[str, Any], Optional[ConfiguredAirbyteCatalog]]:
     parsed_args = AirbyteEntrypoint.parse_args(args)
     config_path, catalog_path = parsed_args.config, parsed_args.catalog
     if parsed_args.command != "read":
         raise ValueError("Only read commands are allowed for Connector Builder requests.")
 
     config = BaseConnector.read_config(config_path)
-    catalog = ConfiguredAirbyteCatalog.parse_obj(BaseConnector.read_config(catalog_path))
+
+    if "__command" not in config:
+        raise ValueError(
+            f"Invalid config: `__command` should be provided at the root of the config but config only has keys {list(config.keys())}"
+        )
+
+    command = config["__command"]
+    if command == "test_read":
+        catalog = ConfiguredAirbyteCatalog.parse_obj(BaseConnector.read_config(catalog_path))
+    else:
+        catalog = None
 
     if "__injected_declarative_manifest" not in config:
         raise ValueError(
             f"Invalid config: `__injected_declarative_manifest` should be provided at the root of the config but config only has keys {list(config.keys())}"
         )
 
-    return config, catalog
+    return command, config, catalog
 
 
-def handle_connector_builder_request(source: ManifestDeclarativeSource, config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog):
-    command = config.get("__command")
+def handle_connector_builder_request(
+    source: ManifestDeclarativeSource, command: str, config: Mapping[str, Any], catalog: Optional[ConfiguredAirbyteCatalog]
```

Comment:
nit: `read_stream` expects `catalog` to be non-None, so we should validate its presence before passing it

Reply:
Ah good call. I didn't see this before I hit merge so will open a new one for that.